src/Workspaces/Remote/ServiceHub/Services/SemanticClassification/ClassificationCachingService.cs

```diff
@@ -21,7 +24,9 @@
 
 namespace Microsoft.CodeAnalysis.Remote
 {
-    internal sealed partial class RemoteSemanticClassificationService : BrokeredServiceBase, IRemoteSemanticClassificationService
+    [ExportWorkspaceService(typeof(IClassificationCachingService))]
+    [Shared]
+    internal class ClassificationCachingService : IClassificationCachingService
```

Comment:
i would be ok sending this in independently as a separate PR as well.  
note: can you keep this sealed if possible?
finally, consider just exporting the type itself.  i'm fine with no interface here.